src/resolvers/commentLikeResolvers.ts

```diff
@@ -1,40 +1,80 @@
-import { GraphQLID, GraphQLList, GraphQLNonNull } from "graphql";
-import { CommentLikeType } from "../types/commentLikeType";
+import {
+  GraphQLID,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLInt,
+  GraphQLObjectType,
+} from "graphql";
 import { CommentLikeModel } from "../models/commentLike";
 import { CommentModel } from "../models/commentModel";
 
+const CommentLikeDetailsType = new GraphQLObjectType({
+  name: "CommentLikeDetails",
+  fields: () => ({
+    count: { type: GraphQLInt }, 
+    likes: { type: new GraphQLList(GraphQLID) }, 
+  }),
+});
+
 export const commentLikeResolvers = {
   Query: {
     getCommentLikes: {
-      type: new GraphQLList(CommentLikeType),
-      args: { blog: { type: new GraphQLNonNull(GraphQLID) } },
+      type: new GraphQLObjectType({
+        name: "CommentLikes",
+        fields: {
+          count: { type: GraphQLInt },
+        },
+      }),
+      args: { comment: { type: new GraphQLNonNull(GraphQLID) } },
       resolve: async (_: any, { comment }: any) => {
-        return await CommentLikeModel.find({ comment })
-          .populate("user")
-          .populate({
-            path: "comment",
-            populate: { path: "user" },
-          });
+        try {
+          const commentData = await CommentModel.findById(comment).select("likes");
+
+          if (!commentData) {
+            throw new Error("Comment not found.");
+          }
+
+          return { count: commentData.likes.length };
+        } catch (error) {
+          console.error("Error fetching comment likes:", error);
+          throw new Error("Failed to fetch comment likes.");
+        }
       },
     },
   },
   Mutation: {
     addCommentLike: {
-      type: CommentLikeType,
+      type: GraphQLInt, 
       args: {
         user: { type: new GraphQLNonNull(GraphQLID) },
         comment: { type: new GraphQLNonNull(GraphQLID) },
       },
       resolve: async (_: any, { user, comment }: any) => {
-        const like = new CommentLikeModel({ user, comment });
-        await CommentModel.findByIdAndUpdate(comment, {
-          $push: { likes: like._id },
-        });
-        const savedLike = await like.save();
-        return (await savedLike.populate("user")).populate({
-          path: "comment",
-          populate: { path: "user" },
-        });
+        try {
+          const existingLike = await CommentLikeModel.findOne({ user, comment });
+          if (existingLike) {
+            throw new Error("User has already liked this comment.");
```

Comment:
If a user has already liked the comment, the like should be removed instead of throwing an error

Reply:
> If a user has already liked the comment, the like should be removed instead of throwing an error

Done, Kindly check